src/components/IntegrationsView/CustomFieldsAndWidgets.tsx

```diff
@@ -36,30 +39,59 @@ export const CustomInputField = ({
   name?: string;
   defaultValue?: string | number;
   placeholder?: string;
+  size?: string;
+  width?: string;
 }) => {
   return (
-    <Box width="100%">
-      <TextField.Root
-        id={id}
-        name={name}
-        type={type}
-        size="2"
-        value={value}
-        defaultValue={defaultValue}
-        placeholder={placeholder}
-      />
+    <Box
+      width={
+        width
+          ? width
+          : size === "short"
+            ? {
+                xs: "50%",
+                initial: "100%",
+              }
+            : "100%"
+      }
+    >
+      {size !== "multiline" ? (
+        <TextField.Root
+          id={id}
+          name={name}
+          type={type}
+          size="2"
+          value={value}
+          variant="soft"
+          color="gray"
+          defaultValue={defaultValue}
+          placeholder={placeholder}
+        />
+      ) : (
+        <TextArea
+          id={id}
+          name={name}
+          size="2"
+          rows={3}
+          value={value}
+          variant="soft"
+          color="gray"
+          defaultValue={defaultValue?.toString()}
+          placeholder={placeholder}
+        />
+      )}
     </Box>
   );
 };
 
 export const CustomLabel = ({
   label,
   htmlFor,
-  width,
+  marginTop,
 }: {
   label: string;
   htmlFor?: string;
-  width?: string;
+  marginTop?: string;
 }) => {
   return (
     <label
```

Comment:
maybe use the `Label` component here

Reply:
`<Text as="label" mt="2" size="2" htmlFor={htmlFor}>`
`mt="2"` is roughly 8px, and `size="2"` is about 14px